### PR TITLE
Eg/create sequelize model user

### PR DESCRIPTION
**Pull request by eigiachero**

### CODE REVIEW

File: sequelize-user-cli.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+npx sequelize-cli model:generate --name User --attributes id:UUID,firstName:string,lastName:string,username:string,salt:string,password:string
```
Comment (abelosorio) [inline sequelize-user-cli.sh]: What is this file? It shouldn't be part of the repo.

File: src/config/database-copy.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (abelosorio) [inline src/config/database-copy.json]: Same with this. Is this a "backup" file?

File: src/db/migrations/20200110001514-create-user.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+'use strict';
```
Comment (abelosorio) [inline src/db/migrations/20200110001514-create-user.js]: Remember that we're using StandarJS (no trailing semicolons).

File: src/db/migrations/20200110001514-create-user.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('Users', {
```
Comment (abelosorio) [inline src/db/migrations/20200110001514-create-user.js]: The table name should be singular and lowercase (another WeDevelop's "practice"). We can talk further on Slack.

File: src/db/migrations/20200110001514-create-user.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      id: {
+        allowNull: false,
+        defaultValue: Sequelize.UUIDV4,
+        primaryKey: true,
+        type: Sequelize.UUID
+      },
+      firstName: {
```
Comment (abelosorio) [inline src/db/migrations/20200110001514-create-user.js:10]: Let's be explicit about the "allowNull" property. Use it always, even when the field allows null values. This is for clarity.

File: src/db/migrations/20200110001514-create-user.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      password: {
+        type: Sequelize.STRING,
+        allowNull: false
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
```
Comment (abelosorio) [inline src/db/migrations/20200110001514-create-user.js]: This field should be nullable. The semantic would be "If updatedAt is null, it means that the field has not been updated".

File: src/db/migrations/20200110001514-create-user.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.dropTable('Users');
```
Comment (abelosorio) [inline src/db/migrations/20200110001514-create-user.js]: Semicolon and table name.

File: src/db/migrations/20200110001514-create-user.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  }
+};
```
Comment (abelosorio) [inline src/db/migrations/20200110001514-create-user.js]: Semicolon and missing eof.

File: src/index.js [hunk 9]
```diff
@@ -29,3 +31,26 @@ httpServer.listen({ port }, () => {
...
+//Sequelize usage test
+const dummy_user = {
+  firstName: "Ezequiel",
+  lastName: "Giachero",
+  username: "Eze123",
+  salt: "",
+  password: "pepito"
+}
+
+async function createShowDestroy(user){
```
Comment (abelosorio) [inline src/index.js]: Some code-style issues. Make sure to set up and run a linter before pushing code.

Comment (abelosorio) [inline src/index.js]: I guess this is a test function, isn't it? If so, make sure to not push it to the repo. If you do need the test function, make sure to add a comment explaining this and why you need it.

File: src/models/index.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Comment (abelosorio) [inline src/models/index.js]: Semicolon.

Comment (abelosorio) [reply src/models/index.js]: SemicolonS

File: src/models/user.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
+import crypto from 'crypto'
+'use strict';
```
Comment (abelosorio) [inline src/models/user.js]: No need to use strict.

File: src/models/user.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+export default (sequelize, DataTypes) => {
+  const User = sequelize.define('User', {
```
Comment (abelosorio) [inline src/models/user.js]: Lowercase.

File: src/models/user.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    id: { type: DataTypes.UUID, allowNull: false, primaryKey: true, defaultValue: DataTypes.UUIDV4 },
```
Comment (abelosorio) [inline src/models/user.js]: This syntax is not very readable. I'd suggest to write it in different lines.

File: src/models/user.js [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+  User.encryptPassword = function (plainPassword, salt) {
+    return crypto.scryptSync(plainPassword, salt, 64).toString('hex')
+  }
+
+  // hooks
+  User.beforeCreate(User.hashPasswordHook.bind(User))
+  User.beforeUpdate(User.hashPasswordHook.bind(User))
+
+  return User;
```
Comment (abelosorio) [inline src/models/user.js]: Semicolons! :D

**Review by abelosorio — CHANGES_REQUESTED** [review]
Looks good overall. Make sure to set up and run standardjs before pushing code.

Great job

File: src/db/migrations/20200110001514-create-user.js [hunk 15]
```diff
@@ -28,16 +29,17 @@ module.exports = {
         allowNull: false
       },
       createdAt: {
-        allowNull: false,
-        type: Sequelize.DATE
+        type: Sequelize.DATE,
+        allowNull: false
+
```
Comment (abelosorio) [inline src/db/migrations/20200110001514-create-user.js]: Please, remove this blank line. Use blank lines only between logic blocks. You can find an explanation for this online, but basically:

> Vertical indents: empty lines for splitting code into logical blocks.
> 
> Even a single function can often be divided into logical blocks. In the example below, the initialization of variables, the main loop and returning the result are split vertically:
> 
```
function pow(x, n) {
  let result = 1;
  //              <--
  for (let i = 0; i < n; i++) {
    result *= x;
  }
  //              <--
  return result;
```
> }
> Insert an extra newline where it helps to make the code more readable. There should not be more than nine lines of code without a vertical indentation.

**Review by abelosorio — COMMENTED** [review]
Much better! :muscle: one more small comment ;)

File: src/db/migrations/20200110001514-create-user.js (hunk 15 above)
Comment (eigiachero) [reply src/db/migrations/20200110001514-create-user.js]: Yes, while I was changing the order to be more consistent I forgot to remove that. Thanks!

**Review by abelosorio — APPROVED** [review]
Great job!

**Review by nahudalla — APPROVED** [review]

**Pull request merged**